.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,131 @@
+name: Benchmark
+
+on:
+  # Run the workflow every day at midnight (00:00) UTC
+  schedule:
+    - cron: "0 0 * * *"
+  # Enable manual trigger for easy debugging
+  workflow_dispatch:
+
+permissions:
+  contents: read
+  pull-requests: write
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+  benchmark:
+    runs-on: "${{ github.repository == 'dagger/dagger' && 'dagger-g2-v0-12-5-4c' || 'ubuntu-latest' }}"
+    timeout-minutes: 10
+
+    strategy:
+      matrix:
+        sdk: [go, python, typescript] # The SDKs you want to test
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install go
+        uses: actions/setup-go@v5
+        if: inputs.dev-engine == 'true'
+        with:
+          go-version: "1.22"
+          cache-dependency-path: "dev/go.sum"
+
+      - name: Install dagger
+        shell: bash
+        env:
+          DAGGER_VERSION: "v0.12.5"
+        run: |
+          if [[ -x "$(command -v dagger)" ]]; then
+            echo "::group::Checking dagger"
+            version="$(dagger --silent version | cut --fields 2 --delimiter ' ')"
+            if [[ "$version" != "$DAGGER_VERSION" ]]; then
+              echo "dagger ${version} is installed, but needed ${DAGGER_VERSION}"
+              exit 1
+            fi
+            echo "::endgroup::"
+          else
+            echo "::group::Installing dagger"
+            curl -fsSL https://dl.dagger.io/dagger/install.sh | BIN_DIR=/usr/local/bin/ sudo -E sh
+            echo "::endgroup::"
+          fi
+
+      - name: Build dev dagger
+        shell: bash
+        run: |
+          echo "::group::Starting dev engine"
+          if ! [[ -x "$(command -v docker)" ]]; then
+            echo "docker is not installed"
+            exit 1
+          fi
+
+          # put env variables in /tmp/local-envs instead of
+          # $GITHUB_ENV to avoid leaking into parent workflow
+          (cd .dagger/mage; go run main.go -w ../.. engine:dev) | tee /tmp/local-envs
```

Comment:
@gerhard Any better of way of doing all of this? I had to copy some bits of the call action

Also -- are those envs path safe? e.g. Is the code below doing a `source` and then `dagger <command>` using the right stuff?

Reply:
I don't think that we need any of this.

We expect this to only run on the `dagger/dagger` repository, which means that we will **always** have access to our CI runners. We can then simplify all of this to use:
- `runs-on: dagger-g2-v0-12-5-4c`
- `runs-on: dagger-g2-main-4c`

That will basically benchmark against the latest released version, and the latest unreleased version. `main` is wired through our entire stack, meaning that the CLI will be automatically installed and made available to the workflow, and the corresponding Engine will be ready to go by the time the workflow starts.